tree/dataframe/inc/ROOT/RDF/RMergeableValue.hxx

```diff
@@ -0,0 +1,458 @@
+subclasses are currently available:
+
+| RMergeableValue Specialization | Corresponding RDataFrame Action(s)         |
+|--------------------------------|--------------------------------------------|
+| RMergeableFill                 | Graph, Histo{1,2,3}D, Profile{1,2}D, Stats |
+| RMergeableGenericAction        | Count, Max, Min, Sum                       |
+| RMergeableMean                 | Mean                                       |
+| RMergeableStdDev               | StdDev                                     |
+*/
+template <typename T>
+class RMergeableValue : public RMergeableValueBase {
+   // Friend function declaration
+   template <typename T1, typename... Ts>
+   friend std::unique_ptr<RMergeableValue<T1>> MergeValues(std::unique_ptr<RMergeableValue<T1>> OutputMergeable,
+                                                           std::unique_ptr<RMergeableValue<Ts>>... InputMergeables);
+   template <typename T1, typename... Ts>
+   friend void MergeValues(RMergeableValue<T1> &OutputMergeable, const RMergeableValue<Ts> &... InputMergeables);
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Aggregate the information contained in another RMergeableValue
+   ///        into this.
+   ///
+   /// Virtual function reimplemented in all the subclasses.
+   ///
+   /// \note All the `Merge` methods in the RMergeableValue family are private.
+   /// To merge multiple RMergeableValue objects please use [MergeValues]
+   /// (namespaceROOT_1_1Detail_1_1RDF.html#af16fefbe2d120983123ddf8a1e137277).
+   virtual void Merge(const RMergeableValue<T> &) = 0;
+
+protected:
+   T fValue; ///< Result of the RDataFrame action wrapped by this mergeable.
+
+public:
+   /**
+      Constructor taking the action result by const reference. This involves a
+      copy of the result into the data member, but gives full ownership of data
+      to the mergeable.
+   */
+   RMergeableValue(const T &value) : fValue{value} {}
+
+   /**
+      Retrieve the value contained in this mergeable object. Meant to be called
+      after merging two ore more objects together to retrieve the final result.
+   */
+   const T &GetValue() const { return fValue; }
+};
+
+/**
+\class ROOT::Detail::RDF::RMergeableGenericAction
+\ingroup dataframe
+\brief Generic specialization of RMergeableValue.
+\tparam T Type of the action result.
+\tparam ActionHelper Action helper class the result was created with.
+
+This specialization of RMergeableValue holds two data members: the result of an
+action and an `std::function` to merge two similar results.
+*/
+template <typename T, typename ActionHelper>
+class RMergeableGenericAction final : public RMergeableValue<T> {
+   /**
+      Alias to the correct `std::function` template instantiation.
+   */
+   using mergefun_t = std::function<T(const T &, const T &)>;
+   const mergefun_t fMergeFun; ///< Merge algorithm held in an `std::function`
```

Comment:
i'd be surprised if std::function was serializable/streamable..?

Reply:
Actually this is true, even before trying it out with pickle there is no support for serialization of `std::function` on the C++ side... damn.

This mean that I probably need to create separate mergeable types also for these simple values and give each of them their own `Merge` method

Reply:
This is solved in the latest commits. Had to add separate specializations for each action, also added default constructors everywhere plus dictionary generation in LinkDef.h